scripts/release_dev.js

```diff
@@ -0,0 +1,109 @@
+const path = require('path');
+
+const semver = require('semver');
+
+const {
+  askForConfirmation,
+  askForText,
+  checkGitStatus,
+  checkNpmAuth,
+  revertCommit,
+  runExec,
+} = require('./release.helpers');
+
+const packageJsonPath = path.resolve(__dirname, '../package.json');
+const packageJson = require(packageJsonPath);
+
+async function checkGitBranchAndStatus() {
+  console.log('Checking your PR branch...');
+  const resultBranch = await runExec('git branch --show-current', {
+    silent: true,
+  });
+  const branchName = resultBranch.stdout.toString().trim();
+  if (branchName === 'main') {
+    console.error(`🟠 You are at "main". Are you sure you wanna release a dev version here?`);
+    process.exit(1);
+  }
+
+  await checkGitStatus();
+}
+
+async function getNewVersion() {
+  const result = await runExec('git rev-parse --short HEAD^');
+  // BUG TODO later The gitSHA does not match the real next commit sha
```

Comment:
I think this is because this command is returning the _next to last_ hash. 

I think `git describe --always` or `git log -1 --abbrev-commit --pretty=oneline` should get you the correct hash?

Reply:
Hum no. `git describe --always` returns the hash of the last commit. Here I want to get the hash of the next (upcoming not yet done) commit. :/

Reply:
Now that I have the approval of both John and Blake, shall we try to create `@0.1.0-dev.xxxx` in this PR or in another PR? 😅 I'd say let's go with this one.

Reply:
> Hum no. `git describe --always` returns the hash of the last commit. Here I want to get the hash of the next (upcoming not yet done) commit. :/

Ah ok. It's not a blocker, but think we should land on something we want to keep for the long-term soon-ish.

Reply:
The alternative, which was my first commit, was to use date instead. eg `0.1.0-dev.20230517172340` (yyyymmddhhmmss). I've seen this in a couple of other libraries in the wild (eg [this one](https://www.npmjs.com/package/@pandacss/node?activeTab=versions)). WDYT?

Reply:
🤔 I kind of like the date + timestamp tbh.

Reply:
I agree. Sorry @brennj I'm turning back to the date again. https://github.com/remoteoss/json-schema-form/pull/3/commits/efe38a131727ea7cd7e2141ae52187d98d5d85a5 Yesterday you asked for the advantages, here's the answer:

- git sha do not match with the upcoming commit. On top of that, they are not historically sortable.
- By using the yyyymmddhhmmss format, we know their order of releases.

Reply:
i'm ok with that ;)